server/src/main/java/com/cloud/server/StatsCollector.java

```diff
@@ -564,11 +657,352 @@ protected Point creteInfluxDbPoint(Object metricsObject) {
+            final OperatingSystemMXBean mxBean = ManagementFactory.getOperatingSystemMXBean();
+            newEntry.setAvailableProcessors(mxBean.getAvailableProcessors());
+            newEntry.setLoadAverage(mxBean.getSystemLoadAverage());
+            if (LOGGER.isTraceEnabled()) {
+                LOGGER.trace(String.format(
+                        "Metrics processors - %d , loadavg - %f ",
+                        newEntry.getAvailableProcessors(),
+                        newEntry.getLoadAverage()));
+            }
+        }
+
+        private void getRuntimeData(@NotNull ManagementServerHostStatsEntry newEntry) {
+            final RuntimeMXBean mxBean = ManagementFactory.getRuntimeMXBean();
+            newEntry.setUptime(mxBean.getUptime());
+            newEntry.setStartTime(mxBean.getStartTime());
+            newEntry.setProcessId(mxBean.getPid());
+            newEntry.setJvmName(mxBean.getName());
+            newEntry.setJvmVendor(mxBean.getVmVendor());
+            newEntry.setJvmVersion(mxBean.getVmVersion());
+            if (LOGGER.isTraceEnabled()) {
+                LOGGER.trace(String.format(
+                        "Metrics uptime - %d , starttime - %d",
+                        newEntry.getUptime(),
+                        newEntry.getStartTime()));
+            }
+        }
+
+        private void getJvmDimensions(@NotNull ManagementServerHostStatsEntry newEntry) {
+            Runtime runtime = Runtime.getRuntime();
+            newEntry.setTotalMemoryBytes(runtime.totalMemory());
+            newEntry.setFreeMemoryBytes(runtime.freeMemory());
+            newEntry.setAvailableProcessors(runtime.availableProcessors());
+            newEntry.setTotalMax(runtime.maxMemory());
+            //long maxMem = runtime.maxMemory();
+            if (LOGGER.isTraceEnabled()) {
+                LOGGER.trace(String.format(
+                        "Metrics proc - %d , maxMem - %d , totalMemory - %d , freeMemory - %f ",
+                        newEntry.getAvailableProcessors(),
+                        newEntry.getTotalMax(),
+                        newEntry.getTotalMemoryBytes(),
+                        newEntry.getFreeMemoryBytes()));
+            }
+        }
+
+        /**
+         * As for data from outside the JVM, we only rely on /proc/ contained data.
+         *
+         * @param newEntry
+         */
+        private void getProcFsData(@NotNull ManagementServerHostStatsEntry newEntry) {
+            String OS = Script.runSimpleBashScript("cat /proc/version");
+            newEntry.setOsDistribution(OS);
+            String mem = Script.runSimpleBashScript("cat /proc/meminfo | grep MemTotal | cut -f 2 -d ':' | tr -d 'a-zA-z '").trim();
+            newEntry.setSystemMemoryTotal(Long.parseLong(mem) * 1024);
+            String free = Script.runSimpleBashScript("cat /proc/meminfo | grep MemFree | cut -f 2 -d ':' | tr -d 'a-zA-z '").trim();
+            newEntry.setSystemMemoryFree(Long.parseLong(free) * 1024);
```

Comment:
OperatingSystemMXBean.getFreePhysicalMemorySize() ?

Reply:
not in java 11 either